src/sets.jl

```diff
@@ -240,3 +239,227 @@ function distance_to_set(d::DefaultDistance, v::AbstractVector{T}, s::MOI.Indica
         (distance_to_set(d, z, MOI.ZeroOne()), distance_to_set(v[2], s.set))
     )
 end
+
+
+# find expression of projections on cones and their derivatives here:
+#   https://stanford.edu/~boyd/papers/pdf/cone_prog_refine.pdf
+
+
+"""
+    projection of vector `z` on zero cone i.e. K = {0} or its dual
+"""
+function projection_on_set(::DefaultDistance, ::MOI.Zeros, z::Array{Float64}, dual=true)
+    return dual ? z : zeros(Float64, size(z))
+end
+
+function projection_on_set(::DefaultDistance, ::MOI.EqualTo, z::Array{Float64}, dual=true)
```

Comment:
EqualTo is a scalar set, so z will be a scalar

Reply:
thanks for pointing out!

Reply:
will remove it